vectorstores/alloydb/vectorestores.go

```diff
@@ -0,0 +1,346 @@
+package alloydb
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/jackc/pgx/v5"
+	"github.com/tmc/langchaingo/embeddings"
+	"github.com/tmc/langchaingo/internal/alloydbutil"
+	"github.com/tmc/langchaingo/schema"
+	"github.com/tmc/langchaingo/vectorstores"
+)
+
+const (
+	defaultIndexNameSuffix = "langchainvectorindex"
+	similaritySearchQuery  = `WITH filtered_embedding_dims AS MATERIALIZED (
+    SELECT
+        *
+    FROM
+        %s
+    WHERE
+        vector_dims (
+                embedding
+        ) = $1
+)
+SELECT
+    data.document,
+    data.cmetadata,
+    (1 - data.distance) AS score
+FROM (
+    SELECT
+        filtered_embedding_dims.*,
+        embedding <=> $2 AS distance
+    FROM
+        filtered_embedding_dims
+    JOIN %s ON filtered_embedding_dims.collection_id=%s.uuid
+    WHERE %s.name='%s') AS data
+WHERE %s
+ORDER BY
+    data.distance
+LIMIT $3`
+)
+
+type VectorStore struct {
+	engine             alloydbutil.PostgresEngine
+	embedder           embeddings.Embedder
+	tableName          string
+	schemaName         string
+	idColumn           string
+	metadataJsonColumn string
+	contentColumn      string
+	embeddingColumn    string
+	metadataColumns    []string
+	overwriteExisting  bool
+	k                  int
```

Comment:
do these need to be on this VectorStore struct? they seem like temporary options